lib/utils.js

```diff
@@ -90,19 +90,18 @@ export const processSingleRepo = (repo) => {
     fields.forEach((field) => {
       data[field] = datapackage[field] || ''
     })
-    data['sample'] = datapackage['sample'] || []
     data['name'] = repo['name'] || ''
     data['id'] = repo['id']
     data['createdAt'] = repo.createdAt
     data['updatedAt'] = repo.updatedAt
 
-    if ('resources' in datapackage) {
+    if ('resources' in datapackage && datapackage['resources'].length > 0) {
       data['resources'] = datapackage['resources']
-      try {
-        data['schema'] = datapackage['resources'][0]['schema'] || []
-      } catch (error) {
-        data['schema'] = { fields: [] }
-      }
+      data['schema'] = datapackage['resources'][0]['schema'] || []
+      data['sample'] = datapackage['resources'][0]['sample']  || datapackage['sample'] || []
```

Comment:
Same comment as for line 48.

Reply:
Same answer. If you can reupload the datasets. I can get rid of that line.

Reply:
Ok, that's great, thank you! I think we can simplify the code further and make sure we don't have to care about extra properties at the top-level. I can re-upload all the datasets, that's not a problem, so we don't have to consider any existing data so far. :slightly_smiling_face:

Reply:
Okay, I'll update the code and push again